### DISCUSSION TITLE
Two Windows Problems

### DISCUSSION THREAD

**Discussion by Stephen-Leibowitz** [pawelsalawa/sqlitestudio#4667] [Q&A]
### Details
I am new to SQLiteStudio, so it is very possible the problems are due to my inexperience. If so, I apologize. When I _Export the database_, I do not initially see the Cancel or Next buttons on the bottom. But if I use my mouse to increase or decrease the window size, even if only horizontally, the missing buttons are displayed and I can use them. It is the same with Tools > Export and Tools > Import. My DPI setting is 120, which Windows labels as Medium: 

![Export](https://user-images.githubusercontent.com/11428233/213887045-f0f22079-ea85-42fc-a2fd-7d87ca20497b.png)

![CustomDPI](https://user-images.githubusercontent.com/11428233/213887069-e8de6eff-14a9-4cf7-80bd-e49ea5e818de.png)

The other problem is that CSV does not appear to be an _Export format_: 

![ExportFormat](https://user-images.githubusercontent.com/11428233/213887119-8f613d2d-ec18-43da-a823-047992655914.png)

![Configuration](https://user-images.githubusercontent.com/11428233/213887126-ff218af8-2a63-43ae-a631-1eb28247bf78.png)

### Operating system
Windows 7

### SQLiteStudio version
3.4.3

**Comment by pawelsalawa** [comment]
Ad. 1) I think it's a problem only under Windows 7, because Qt (a toolkit which SQLiteStudio uses for UI) no longer officially supports Windows 7. I've made a test on Windows 10 with scaling up entire interface and I don't observe such issues.

Ad. 2) CSV format is not capable of exporting whole database, because there are indexes and triggers, and most importantly - multiple tables (at least there could be). The CSV format is not capable of holding data of different formats in the same output file. That's why CSV format is available only if single-table export is being done. To do it this way you need to open the table (double click on it) and from the Table Window there is an "Export" button there, which will export only this table. Then you should see CSV format on the list.

**Reply by Stephen-Leibowitz** [reply]
I appreciate the clarification on CSV exporting.

Concerning the button problem. There are SQLiteStudio windows that do not have the problem. They have a subtle visual difference from the windows with the problem:

![AddADatabase](https://user-images.githubusercontent.com/11428233/213932147-c4ffbd99-5b09-4b5a-b156-22312cbf326e.png)

![Export](https://user-images.githubusercontent.com/11428233/213932166-6dcbf879-91d3-4951-b4a5-cea6c4e13062.png)

![Export2](https://user-images.githubusercontent.com/11428233/213932178-544e3052-416c-41a8-b934-29bc814f8d2d.png)

The second window is Export, where the buttons Cancel and Next are missing. I then used the mouse to slightly decrease the width, which is shown in the third window. The resizing revealed the buttons. The background of the bottom section of Export, in which the buttons are located, is slightly darker than the background of the upper section. In contrast, the background of (Add a) Database is the same color throughout the window, except the title area and border.

I tested SQLiteStudio on a different Windows 7 machine at both 120 and 96 DPI. The problem was present at both DPI settings. Also, [Windows 7 appears to be a supported platform in Qt 5.15](https://doc.qt.io/qt-5/supported-platforms.html), but not in Qt 6.

**Reply by pawelsalawa** [reply]
You're right. Windows 7 is still supported with this Qt version. I remembered it incorrectly.

Nevertheless, SQLiteStudio 3.5.0 is going to migrate to Qt 6 and I'm not sure if it will even run on Windows 7 at all. I'm not really convinced to fix something for the system, that soon SQLiteStudio will not even run on.
I don't even have Windows 7 to test this on and to fix it.

I don't think the benefit is worth the effort. Sorry.
